src/urllib3/poolmanager.py

```diff
@@ -353,7 +353,7 @@ def connection_from_pool_key(
         return pool
 
     def connection_from_url(
-        self, url: str, pool_kwargs: Optional[Dict[str, Any]] = None
+        self, url: Optional[str], pool_kwargs: Optional[Dict[str, Any]] = None
```

Comment:
Does `connection_from_host` has any chance of succeeding with an empty URL? If not, then it seems better to prevent None here.

Reply:
With empty `Url`, the `host` attr will be `None`. by passing `None` to `connection_from_host`, it will raise `LocationValueError`
https://github.com/urllib3/urllib3/blob/d8eabf304190c62766e05c79ef44be7ff94cfbcb/src/urllib3/poolmanager.py#L293

Also, `connection_from_host` accepts `None`:
https://github.com/urllib3/urllib3/blob/d8eabf304190c62766e05c79ef44be7ff94cfbcb/src/urllib3/poolmanager.py#L277

Reply:
Got it. The Optional looks fine to me then, but the question I'm asking myself is which type would be more useful here? I don't think anyone wants to pass a special value (None) which will immediately raise -- presumably anyone who is using typing would prefer mypy to warn them about it already instead of raising at runtime.

If what I'm saying makes sense, the `if None raise LocationValueError(...)` check can still remain, but only as a runtime sanity check (and will probably require ignoring an `unreachable` mypy error *inside* the function, but that's an expected ignore).

Reply:
Yeah, I think we can revert this change and ignore mypy check it test.